linode/manager
# feat: [M3-6811] - Add AGLB Service Target landing page

## Description 📝
Build the AGLB service targets landing page according to the mockup. Since this page remained the same in the Beta mock and all follow up tasks are documented in another ticket, I think it's safe to review this now.

TODO:
- [x] Make ticket(s?) for follow up items - see M3-6882
- [x] Any clean up

## Major Changes 🔄
- Builds skeleton of table and table rows
- Add a RQ query for GET /v4/aglb/service-targets and uses it 
- Adds basic unit tests 

🐛  Fixes for a couple things not directly related to this PR (see self-review):
- Renames the files and initial routes from `entrypoints` to `service-targets` (typo in M3-6822 ticket)
- Fixes a routing issue on the Load Balancer details page so that tabs are correctly selected
- Fixes a routing issue in #9405 where the create menu was not routing to the create load balancer page

## Preview 📷
![Screenshot 2023-07-14 at 9 11 10 AM](https://github.com/linode/manager/assets/114685994/2bc43a2b-0092-4354-aadd-744779ddcc19)

## How to test 🧪
1. **How to setup test environment?**
Check out this branch, toggle the MSW on, and toggle the AGLB feature flag on. 
2. **How to verify changes?**
- Navigate to http://localhost:3000/loadbalancers/service-targets
- Verify that table appears with mock service target data and it matches the beta mocks. (Note: endpoints are still TBD)
- Verify that loading spinner appears when table is loading.
- Block network requests to service-targets endpoint in browser console (or mock error response in MSW: `res(ctx.status(500), ctx.json({}));`) and verify that table renders error state. 

Also please verify 🐛 fixes:
- Visiting http://localhost:3000/loadbalancer/1/settings takes you to the loadbalancer details page with the Settings tab selected. Same goes for visiting any of the tabs via URL. Navigating to http://localhost:3000/loadbalancer/1 should select the Summary tab by default.
- Clicking on Global Load Balancer in the Create menu should navigate to https://localhost:3000/loadbalancers/create (not  the details page).
3. **How to run Unit or E2E tests?**
```
yarn test ServiceTarget
```
Note that all the tests for `ServiceTargetActionMenu` are skipped for now until further UX direction for M3-6882. Some tests for `ServiceTargetLanding` are still not complete and will be finished in M3-6882.
